go/vt/logutil/logger.go

```diff
@@ -246,6 +246,12 @@ func (ml *MemoryLogger) Clear() {
 	ml.mu.Unlock()
 }
 
+func (ml *MemoryLogger) LogEvents() []*logutilpb.Event {
+	ml.mu.Lock()
+	defer ml.mu.Unlock()
+	return ml.Events
```

Comment:
If the slice can be mutated or cleared concurrently you may want to return `slices.Clone(ml.Events)` so that the returned backing array shares no data with the one owned by MemoryLogger.

Reply:
Ah yeah, would be safer here indeed.